llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -612,6 +614,10 @@ AMDGPUTargetMachine::AMDGPUTargetMachine(const Target &T, const Triple &TT,
                         FS, Options, getEffectiveRelocModel(RM),
                         getEffectiveCodeModel(CM, CodeModel::Small), OptLevel),
       TLOF(createTLOF(getTargetTriple())) {
+  // FIXME: There are some scenarios where targets may not have hardware traps,
+  // and external calls to `abort` also fail. For now, do a blanket-disable.
+  if (!EnableTrapUnreachable.getNumOccurrences())
+    this->Options.TrapUnreachable = false;
```

Comment:
Targets should not have to flip bits in this setting like this

Reply:
This is already happening all over the place though. See PS4, PS5, WASM, X86, AArch64, etc.